dev/fattest.simplicity/src/componenttest/containers/ExternalTestServiceDockerClientStrategy.java

```diff
@@ -302,15 +308,65 @@ protected boolean isApplicable() {
         return true;
     }
 
-    public static boolean useRemoteDocker() {
-        if (USE_REMOTE_DOCKER) {
-            return true; // remote docker explicitly requested
-        }
-        if (Boolean.parseBoolean(System.getenv("GITHUB_ACTIONS"))) {
-            return false; // always use local docker for GH Actions
-        }
-        // Otherwise, use local docker for local runs, and remote docker for remote (RTC) runs
-        return !FATRunner.FAT_TEST_LOCALRUN;
-    }
+    /**
+     * Determines if we are going to attempt to run against a remote
+     * docker host, or a local docker host.
+     *
+     * Priority:
+     * 1. System Property: fat.test.use.remote.docker
+     * 2. System Property: fat.test.docker.host -> REMOTE
+     * 3. System: GITHUB_ACTIONS -> LOCAL
+     * 4. System: WINDOWS -> REMOTE
+     *
+     * default (!!! fat.test.localrun)
+     *
+     * @return true, we are running against a remote docker host, false otherwise.
+     */
+    private static boolean useRemoteDocker() {
+        boolean result;
+        String reason;
+
+        do {
+            //State 1: fat.test.use.remote.docker should always be honored first
+            if (System.getProperty("fat.test.use.remote.docker") != null) {
+                result = Boolean.getBoolean("fat.test.use.remote.docker");
+                reason = "fat.test.use.remote.docker set to " + result;
+                break;
+            }
+
+            //State 2: User provided a remote docker host, assume they want to use the remote host
+            if (USE_DOCKER_HOST != null) {
+                result = true;
+                reason = "fat.test.docker.host set to " + USE_DOCKER_HOST;
+                break;
+            }
 
+            //State 3: Github actions build should always use local
+            if (Boolean.parseBoolean(System.getenv("GITHUB_ACTIONS"))) {
```

Comment:
If Github actions should always be built local, shouldn't this be the first thing checked?  Or is there a possible scenario where you would want to use a remote docker with Github Actions?

Reply:
Yes and no.  In my opinion if someone sets a property explicitly, then they should expect it to be honored.  So if this came first, we would not honor fat.test.use.remote.docker or fat.test.docker.host in some cases. 

As a note both of these properties are user provided, not something we calculate at build time.

Reply:
Ok, I thought you were determining if it was a GHA from the environment somehow.  Thanks for explaining that to me.